controller/nodegroup.go

```diff
@@ -327,3 +327,50 @@ func getImageRootDeviceName(imageID *string, ec2Service *ec2.EC2) (*string, erro
 
 	return describeOutput.Images[0].RootDeviceName, nil
 }
+
+// getNodegroupConfigUpdate returns an UpdateNodegroupConfigInput that represents desired state and a bool
+// indicating whether an update needs to take place to achieve the desired state.
+func getNodegroupConfigUpdate(clusterName string, ng eksv1.NodeGroup, upstreamNg eksv1.NodeGroup) (eks.UpdateNodegroupConfigInput, bool) {
```

Comment:
Is this returning a non-pointer `eks.UpdateNodegroupConfigInput` because of the tests?

Reply:
I'm returning a value because of the test and I'm not returning a pointer so that the value goes on the stack and not the heap. My understanding is it's best to do this unless you have a good reason to use a pointer as not using one should lead to less bugs and slightly better performance. I'm probably not super consistent with it and it's not a huge optimization but it's something that was discussed in some go classes we took from Ardan labs.

Reply:
I know we only broke it out for testing, but I'm always happy when a functional block gets put into a separate method. IMO we have too many run-on functions.